apps/ml-service/src/main.py

```
@@ -210,11 +211,87 @@ def forecast_drug(
+            "trend_factor": trend_factor,
+            "adaptive_predictions": predictions,
+            "summary": {
+                "total_predicted_7_days": sum(p['predicted_demand'] for p in predictions[:7]),
+                "total_base_prediction": sum(p['base_prediction'] for p in predictions[:7]),
+                "average_adjustment": sum(p['adjustment_applied'] for p in predictions[:7]) / len(predictions[:7])
+            },
+            "generated_at": datetime.now().isoformat()
+        }
+        
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)